Replace javax.validation with jakarta.validation

Due to such breaking change, also change the version to 2.0.0

This is most likely the cause why it can only be build with Java 8, which is an undesired requirement nowadays:

```
[INFO] --- maven-compiler-plugin:3.11.0:compile (default-compile) @ packageurl-java ---
[INFO] Changes detected - recompiling the module! :source
[INFO] Compiling 7 source files to D:\Data\java-workspace\github\packageurl-java\target\classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] D:\Data\java-workspace\github\packageurl-java\src\main\java\com\github\packageurl\validator\PackageURLConstraintValidator.java:[32,8] cannot access java.lang.Object
  bad class file: /modules/java.base/java/lang/Object.class
    class file has wrong version 55.0, should be 53.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
[INFO] 1 error
```

--- a/pom.xml
+++ b/pom.xml
@@ -87,9 +87,9 @@
 
     <dependencies>
         <dependency>
-            <groupId>javax.validation</groupId>
-            <artifactId>validation-api</artifactId>
-            <version>2.0.1.Final</version>
+            <groupId>jakarta.validation</groupId>
+            <artifactId>jakarta.validation-api</artifactId>
+            <version>3.1.1</version>
             <optional>true</optional>
             <scope>provided</scope>
         </dependency>

--- a/src/main/java/com/github/packageurl/validator/PackageURL.java
+++ b/src/main/java/com/github/packageurl/validator/PackageURL.java
@@ -21,8 +21,8 @@
  */
 package com.github.packageurl.validator;
 
-import javax.validation.Constraint;
-import javax.validation.Payload;
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;

--- a/src/main/java/com/github/packageurl/validator/PackageURLConstraintValidator.java
+++ b/src/main/java/com/github/packageurl/validator/PackageURLConstraintValidator.java
@@ -22,8 +22,8 @@
 package com.github.packageurl.validator;
 
 import com.github.packageurl.MalformedPackageURLException;
-import javax.validation.ConstraintValidator;
-import javax.validation.ConstraintValidatorContext;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
 
 /**
  * A JSR-303 compliant validator that validates String fields conform to the Package URL specification.

